### PR TITLE
Modified Dam project to support Data dependent routing.

### PR DESCRIPTION
**Pull request by supragyan**
Modified below files to support Data dependent routing in Dam project.  …
- MyFcBaseLogic.cs
- DamSqlSvr.cs

### CODE REVIEW

File: root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs [hunk 1]
```diff
@@ -280,6 +298,11 @@ protected override void UOC_DoAction(BaseParameterValue parameterValue, ref Base
 
                 if (dam != null)
                 {
+                    if (parameterValue.ActionType.Split('%')[0] == "SqlDbWithDataDependent")
+                    {
+                        // Azure SQL用のDamを生成
+                        dam = new DamSqlDbWithMultiShard.DamSqlDbWithDataDependentRouting();
```
Comment (daisukenishino) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs]: @supragyan 

I couldn't find implementation of the DamSqlDbWithDataDependentRouting.

File: root_VS2012/programs/C#/Frameworks/Infrastructure/Public/Db/DamSqlSvr.cs [hunk 2]
```diff
@@ -55,20 +55,10 @@
```
Comment (daisukenishino) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Public/Db/DamSqlSvr.cs]: @supragyan 

This modification does not require to DamSqlSvr.
However, I think that it is necessary to correct the accessor for passing the connection object.

File: root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs (hunk 1 above)
Comment (daisukenishino) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs]: @supragyan 

And, don't perform the modifications to the MyFcBaseLogic at this branch.
Should perform modifitication of the MyFcBaseLogic at the "damframework-chngs" branch.

- see : https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/152

Comment (supragyan) [inline root_VS2012/programs/C#/Frameworks/Infrastructure/Business/Business/MyFcBaseLogic.cs]: @daisukenishino 
\>Should perform modifitication of the MyFcBaseLogic at the "damframework-chngs" branch.
We created new [153 pullrequest](https://github.com/SymphonyTeleca/OpenTouryoTemplates/pull/153) which contains only Business project changes.

Please review and let us know.

**Pull request closed without merging**